### PR TITLE
Allow sampling and redaction rules for Scalyr

### LINKED ISSUE #59: Scalyr: Allow sampling rules for containers
Add support for [Scalyr sampling rules](https://eu.scalyr.com/help/scalyr-agent#filter) to Pod containers. The implementation should be similar to Scalyr parsers support.

Example:

```
kubernetes-log-watcher/scalyr-sampling-rules: '[{"container": "my-container", "sampling-rules":[{ match_expression: "<expression here>", sampling_rate: 0 }]}]'
```

### PR DESCRIPTION
**Pull request by femueller**
This feature enables using sampling and redaction rules for Scalyr.

More details about sampling_rules and redaction_rules can be found
here:
https://eu.scalyr.com/help/scalyr-agent

This fixes #59

### CODE REVIEW

**Comment by femueller** [comment]
:+1:

**Comment by femueller** [comment]
:+1:

File: README.rst [hunk 1]
```diff
@@ -321,6 +321,29 @@ The default parser for container logs is ``json`` parser. In some cases however
...
 The value of ``kubernetes-log-watcher/scalyr-parser`` annotation should be a json serialized list. If ``container`` value did not match, then default parser is used (i.e. ``json``).
 
+Scalyr sampling rules
+....................
+
+Sampling rules enable to only ship a certain pattern that matches a regular expression and specified amount of log percentage to Scalyr. The example shows an expression that matches ``app-1`` and a match expression ``my-expression``. If it's met, only 10% of it will be shipped to Scalyr using a ``sampling_rate`` of ``0.1``.
+
+.. code-block:: yaml
+
+  sampling rules:
```
Comment (mohabusama) [inline README.rst]: s/sampling rules/annotations

File: README.rst [hunk 2]
```diff
@@ -321,6 +321,29 @@ The default parser for container logs is ``json`` parser. In some cases however
...
+
+
+Scalyr log redaction
+....................
+
+Redaction rules enable to avoid shipping sensitive data that shouldn't get transferred to Scalyr either getting fully removed from log files or to replace them with specific strings. The first example below shows how matches will be fully removed and the second shows how matches will be replaced with a different string.
+
+.. code-block:: yaml
+
+  redaction rules:
```
Comment (mohabusama) [inline README.rst]: s/redaction rules/annotations

**Comment by femueller** [comment]
:+1:

**Comment by mohabusama** [comment]
👍

**Pull request merged**